include/rocksdb/options.h

```diff
@@ -926,6 +926,9 @@ struct DBOptions {
   // relies on manual invocation of FlushWAL to write the WAL buffer to its
   // file.
   bool manual_wal_flush = false;
+
+  // If true flushes all column families atomically.
+  bool atomic_flush = false;
```

Comment:
Is it possible to avoid a new option? Can we say that if the user has at least one outstanding write with disableWAL=true, we issue atomic flush? We have a flag for that: https://github.com/facebook/rocksdb/blob/v5.12.2/db/db_impl.cc#L255 (although the flag can be improved).

Reply:
I think this is feasible.
If we do this, then atomic flush is not optional if `disableWAL=true`.

Reply:
Isn't that a big change to the default behavior that not everybody would want? Atomic CF flush can cause small L0 files if ingest rate is uneven across CFs, which leads to higher write-amp.

Reply:
Yes, the change is not going to be very small, in comparison to the approach with an additional option. The question is whether `disableWAL=true` and `atomic_flush=true` is a common combination. If we see an overwhelming demand for this feature, then it makes sense not to use the `atomic_flush` option. Another thing to note is that we may not want the number of options to increase.
Right now, I'm going to proceed with original approach, and will come back to revisit this design choice afterwards.

Reply:
I'm late to this review.. by original you mean adding new option?

do we have notion of experimental option? the one that can be added but not be declared as official part of public API, and be removed easily after if not promoted to stable option (assuming that people who opt in to use it are fine with not having backward compatibility)?

Reply:
Thanks @mikhail-antonov for your comment.
Yes, by 'original' I mean adding an option called 'atomic_flush'. It is disabled by default. Therefore, if not needed, the application does not need to change.